### PR TITLE
Limit external grader result images to width of context

### PR DESCRIPTION
**Pull request by jonatanschroeder**
As discussed on Slack. Allows the images generated by external graders to have custom styles (e.g., custom size, rotation, border, etc.). Also updates the documentation for this feature to clarify the available functionality and limitations.

### CODE REVIEW

File: docs/externalGrading.md
Comment (jonatanschroeder) [inline docs/externalGrading.md:177]: Removed since it's a duplicate of the text two paragraphs above.

**Comment by nwalters512** [comment]
Is this actually necessary? I'm generally wary about letting folks inject arbitrary styles into our elements, as that makes it difficult for us to confidently make changes to elements themselves. If our defaults are faulty in some way, let's fix those instead of requiring instructors to override styles.

**Comment by echuber2** [comment]
The issue is that if a very large image is returned, it doesn't display well. If nothing else, I think the default max width shouldn't overflow the width of the enclosing div.

**Comment by nwalters512** [comment]
> If nothing else, I think the default max width shouldn't overflow the width of the enclosing div.

This is exactly the kind of change we should make. Can we slap an [`img-fluid`](https://getbootstrap.com/docs/4.6/content/images/#responsive-images) on the `` tag and call it a day, or are there other changes we should look at too?

**Comment by jonatanschroeder** [comment]
I don't mind restricting it to width either using classes or style. My thought of adding a generic style was to give flexibility, but I see your point. I'll make changes.

**Comment by nwalters512** [comment]
I appreciate it. I know PL's whole thing is flexibility, but I think this is a case where constraints are actually overall good.

**Comment by jonatanschroeder** [comment]
@nwalters512 done. I've settled for the img-fluid suggestion for simplicity.

**Comment by eliotwrobson** [comment]
I think this should wait until #7977 is merged, since that is a pretty large change to this element and is mostly ready to go. It should be easier to rebase / test this element with this change once that PR is in.

**Comment by jonatanschroeder** [comment]
> I think this should wait until #7977 is merged, since that is a pretty large change to this element and is mostly ready to go. It should be easier to rebase / test this element with this change once that PR is in.

There is no file in common between these two PRs. #7977 changes the .py file, this one changes the .mustache file.

File: apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.mustache
Comment (nwalters512) [inline apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.mustache:53]: Nit: don't let this stand in the way of merging, but we could use a `<figure>` element to associate the label (caption) with the image for improved accessibility: https://getbootstrap.com/docs/5.3/content/figures/

**Review by nwalters512 — APPROVED** [review]

File: apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.mustache
Comment (jonatanschroeder) [reply apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.mustache:54]: Indeed, there was no need to do the test since a null label causes no output in mustache.

File: apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.mustache
Comment (jonatanschroeder) [reply apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.mustache]: Done in 9b94238f8. I didn't use the Bootstrap `figure-caption` class as I wanted to keep the same label formatting used in the rest of the element (the merits of that can be discussed in a separate PR).

**Pull request merged**
